crates/biome_js_analyze/src/lint/correctness/no_private_imports.rs

```diff
@@ -276,15 +276,13 @@ fn get_restricted_imports_from_module_source(
     node: &JsModuleSource,
     options: &GetRestrictedImportOptions,
 ) -> SyntaxResult<Vec<NoPrivateImportsState>> {
-    let path: Box<str> = options.target_path.as_str().into();
```

Comment:
This allocation was done unconditionally on every static import we encountered. No more :)